tests/unittest_test.sh

```diff
@@ -94,4 +104,11 @@ function test_basic_failing_test() {
   expect_log "In test _basic_failing_test from //tests:unittest_tests.bzl: Expected \"1\", but got \"2\""
 }
 
+function test_fail_unexpected_passing_test() {
+  ! bazel test //testdir:fail_unexpected_passing_test --test_output=all --verbose_failures \
```

Comment:
Is the exclamation mark intentional? What does it do?

Reply:
It is! It is for testing intentional failures. exclamation mark is a negation of return code.
there are two types of tests.
1.  bazel test <target> || fail "should've passed"    (the fail only occurs if bazel test failed)
2. ! bazel test <target> || fail "should've failed" (the fail only occurs if the bazel test succeded)